src/kiero/kiero.cpp

```diff
@@ -536,8 +562,17 @@ kiero::Status::Enum kiero::init(RenderType::Enum _renderType)
 				commandList->Release();
 				commandList = NULL;
 
-				swapChain->Release();
-				swapChain = NULL;
+				if (swapChain != NULL)
```

Comment:
Could just use a ComPtr for direct3d types

Reply:
Sure. But I saw that all the many other `IUnknown` derived types in kiero are also bare pointers, and I just went along with that in order not to disturb the peace of the file :wink:.

I can of course change this to a ComPtr, but in that case I'd prefer to make the change consistent across the entire file, and then make a new PR for that change. Thoughts?